`HttpDownloaderTests` are stuck on macOS

Mostly seen on the macOS runners.

https://github.com/Nexus-Mods/NexusMods.App/actions/runs/8263328851/attempts/1?pr=1059

```
2024-03-13T10:45:43.6695920Z  Total time: 3.2063 Seconds
2024-03-13T10:45:43.6748630Z     76>Done Building Project "/Users/runner/work/NexusMods.App/NexusMods.App/tests/Extensions/NexusMods.BCL.Extensions.Tests/NexusMods.BCL.Extensions.Tests.csproj" (VSTest target(s)).
2024-03-13T10:45:51.1888280Z   Passed NexusMods.Networking.HttpDownloader.Tests.HttpDownloaderTests.CanDownloadFromLocalServer [112 ms]
2024-03-13T13:16:56.3365370Z ##[error]The operation was canceled.
2024-03-13T13:16:56.3862970Z Post job cleanup.
```

The test `CanDownloadFromLocalServer` works fine, but the next one stalls the entire test. It can't be `CanDownloadFromExternalSource` since that requires networking, which the CI doesn't have (it filters out `RequiresNetworking`).

--- a/tests/Networking/NexusMods.Networking.HttpDownloader.Tests/AdvancedHttpDownloaderTests.cs
+++ b/tests/Networking/NexusMods.Networking.HttpDownloader.Tests/AdvancedHttpDownloaderTests.cs
@@ -52,6 +52,7 @@ public class AdvancedHttpDownloaderTests
     }
 
     [Fact]
+    [Trait("RequiresNetworking", "True")]
     public async Task CanResumeDownload()
     {
         await using var path = _temporaryFileManager.CreateFile();

